shell/key-bindings.fish

```diff
@@ -84,4 +86,26 @@ function fzf_key_bindings
     bind -M insert \cr fzf-history-widget
     bind -M insert \ec fzf-cd-widget
   end
+
+  function __fzf_get_starting_dir
+    set -l dir (commandline -t)
+
+    # Default to current dir if no argument input
+    if [ -z "$dir" ]
+        set dir '.'
+    end
+
+    # The commandline token might be escaped, we need to unescape it.
+    set dir (eval "printf '%s' $dir")
+
+    if [ ! -d "$dir" ]
+      return 1
```

Comment:
Your point is valid and I can surely understand it. I'm just concerned that there might be users who will not [appreciate the change](https://xkcd.com/1172/).

Anyway, not showing any result is surely not the best feedback we can give to the users in that scenario. One thing we can consider is to use the last token, that is not directory, as the default `--query` of the fzf command. That's how fuzzy completion of bash and zsh currently works, for example:

```sh
# Scanning ~/github directory with the default query set to "non-existent"
vim ~/github/non-existent**<tab>

# Scanning current directory with the default query "non-existent"
vim non-existent**<tab>
```

If this apply this scheme to `alt-c`

```sh
git add ../test<alt-c>
  # This will open fzf finder scanning the parent directory with "test" as the default query
  # Then we select a directory (like tests) and the current directory is changed
  # and we're left with "git add " in the command line

git add<alt-c>
  # This will start fzf with "add" as the default query, which is probably
  # not the intention of the user, but at least the user knows what's going on.
```

I'm actually not familiar with fish scripting and not sure if it's easy to implement this in fish. What do you think?